File: ProjectJava/src/main/java/com/example/projectjava/Repository/DonorDatabaseRepository.java
package com.example.projectjava.Repository;

import com.example.projectjava.Domain.Donor;
import com.example.projectjava.InterfaceRepositories.DonorRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class DonorDatabaseRepository implements DonorRepository {
    private final JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public DonorDatabaseRepository(Properties properties) {
        logger.info("Initializing DonorDatabaseRepository with properties: {}",properties);
        dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Iterable<Donor> getAll() {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        List<Donor> donorsList = new ArrayList<>();
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM donor;")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    Integer donorId = result.getInt("id");
                    String address = result.getString("address");
                    String phone = result.getString("phone");
                    String name = result.getString("name");
                    Integer donated_sum = result.getInt("donated_sum");
                    Donor donor = new Donor(name,phone,address,donated_sum);
                    donor.setId(donorId);
                    donorsList.add(donor);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return donorsList;
    }

    @Override
    public Donor getOne(Integer id) throws Exception {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        Donor  donor=null;
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM donor WHERE donor.id = " + id + ";")){
            try(ResultSet result = preparedStatement.executeQuery()){
                if(result.next()){
                    String address = result.getString("address");
                    String phone = result.getString("phone");
                    String name = result.getString("name");
                    Integer donated_sum = result.getInt("donated_sum");
                    donor = new Donor(name,phone,address,donated_sum);
                    donor.setId(id);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return donor;
    }

    @Override
    public Donor add(Donor entity) throws Exception {
        logger.traceEntry();
        Connection connection= dbUtils.getConnection();
        try(PreparedStatement preparedStatement=connection.prepareStatement("INSERT INTO donor(name,phone,address,donated_sum) VALUES (?,?,?,?);")){
            preparedStatement.setString(1,entity.getName());
            preparedStatement.setString(2,entity.getPhone());
            preparedStatement.setString(3,entity.getAddress());
            preparedStatement.setInt(4,entity.getDonated_sum());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public Donor remove(Integer id) throws Exception {
        logger.traceEntry("Removing donor with id {}",id);
        Donor donor = getOne(id);
        Connection connection= dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("DELETE FROM donor WHERE donor.id = " + id+ ";")){
            int result = preparedStatement.executeUpdate();
            logger.trace("Deleted {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        return donor;
    }

    @Override
    public Donor update(Integer id, Donor newEntity) throws Exception {
        logger.traceEntry("Modifying donor with id  {}",id);
        Donor oldDonor = getOne(id);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("UPDATE donor SET " +
                "name = '" + newEntity.getName() +
                "',phone = '"+ newEntity.getPhone()+
                "',address = '" + newEntity.getAddress() +
                "',donated_sum = '" + newEntity.getDonated_sum()+
                "'  WHERE donor.id ='" + id + "';"
        )){
            int result = preparedStatement.executeUpdate();
            logger.trace("Modified {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return oldDonor;
    }

    @Override
    public Donor getDonor(String name, String phone, String address, Integer donated_sum) throws Exception {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        Donor  donor=null;
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM donor WHERE donor.name = '" + name +
                "'AND donor.phone = '" +phone +"' AND donor.donated_sum = '"+ donated_sum+ "';")){
            try(ResultSet result = preparedStatement.executeQuery()){
                if(result.next()){
                    Integer id = result.getInt("id");
                    donor = new Donor(name,phone,address,donated_sum);
                    donor.setId(id);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return donor;
    }
}

File: ProjectC#/WindowsFormsApp/WindowsFormsApp/Repository/DonorDatabaseRepository.cs
using ConsoleApp.Domain;
using ConsoleApp.InterfaceRepository;
using log4net;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp.Repository
{
    public class DonorDatabaseRepository : DonorRepository
    {
        private static readonly ILog Log = LogManager.GetLogger("DonorDatabaseRepository");

        public DonorDatabaseRepository()
        {
            Log.Info("Creating DonorDatabaseRepository");
        }

        public List<Donor> findAll()
        {
            Log.Info("Entering GetAll");

            var donors = new List<Donor>();
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from donor;";
                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        var id = dataReader.GetInt32(dataReader.GetOrdinal("id"));
                        var phone = dataReader.GetString(dataReader.GetOrdinal("phone"));
                        var name = dataReader.GetString(dataReader.GetOrdinal("name"));
                        var address = dataReader.GetString(dataReader.GetOrdinal("address"));
                        var donated_sum = dataReader.GetInt32(dataReader.GetOrdinal("donated_sum"));
                        var donor = new Donor(name, phone, address, donated_sum);
                        donor.setId(id);
                        donors.Add(donor);
                    }
                }
            }
            Log.InfoFormat("Exiting GetAll with values {0}", donors);
            return donors;
        }

        public Donor findOne(int id)
        {
            Log.InfoFormat("Entering GetOne with value {0}", id);

            Donor donor = null ;
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from donor where donor.id = " + id + ";";
                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        var phone = dataReader.GetString(dataReader.GetOrdinal("phone"));
                        var name = dataReader.GetString(dataReader.GetOrdinal("name"));
                        var address = dataReader.GetString(dataReader.GetOrdinal("address"));
                        var donated_sum = dataReader.GetInt32(dataReader.GetOrdinal("donated_sum"));
                        donor = new Donor(name, phone, address, donated_sum);
                        donor.setId(id);
                    }
                }
            }
            return donor;
        }

        public void save(Donor donor)
        {
            Log.InfoFormat("Entering Add with value {0}", donor);

            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText =
                    "INSERT INTO donor(name,phone,address,donated_sum) VALUES (@name,@phone,@address,@donated_sum);";

                var dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@name";
                dataParameter.Value = donor.getName();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@phone";
                dataParameter.Value = donor.getPhone();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@address";
                dataParameter.Value = donor.getAddress();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@donated_sum";
                dataParameter.Value = donor.getDonated_sum();
                command.Parameters.Add(dataParameter);

                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting getOne with value {0}", donor);
        }

        public void update(int id, Donor newEntity)
        {
            Log.InfoFormat("Entering Modify with value {0}", id);

            var oldDonor = findOne(id);
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "UPDATE donor SET " +
                "name = '" + newEntity.getName() +
                "',phone = '" + newEntity.getPhone() +
                "',address = '" + newEntity.getAddress() +
                "',donated_sum = '" + newEntity.getDonated_sum() +
                "'  WHERE donor.id ='" + id + "';";
                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting Remove with value {0}", oldDonor);
        }
        public void delete(int id)
        {
            Log.InfoFormat("Entering Remove with value {0}", id);

            var donor = findOne(id);
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "DELETE FROM donor WHERE donor.id = " + id + ";";
                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting Remove with value {0}", donor);
        }

        public Donor getDonor(string name, string phone, string address, int donated_sum)
        {
            Donor donor = null;
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from donor where donor.phone = '" + phone + "' and donor.donated_sum = " + donated_sum + ";";
                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        var id = dataReader.GetInt32(dataReader.GetOrdinal("id"));
                        donor = new Donor(name, phone, address, donated_sum);
                        donor.setId(id);
                    }
                }
            }
            return donor;
        }


    }
}
